Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -900,6 +904,31 @@ extension SourceKitServer {
       }
       self.workspaces.append(contentsOf: newWorkspaces)
     }
+
+    // For each document that has moved to a different workspace, close it in
+    // the old workspace and open it in the new workspace.
+    for docUri in self.documentManager.openDocuments {
+      let oldWorkspace = preChangeWorkspaces[docUri]
+      let newWorkspace = self.workspaceForDocument(uri: docUri)
+      if newWorkspace !== oldWorkspace {
+        guard let snapshot = documentManager.latestSnapshot(docUri) else {
+          continue
+        }
+        if let oldWorkspace {
```

Comment:
This syntax is not supported by the released version of swift (5.6.1), only by 5.7+.  We should keep using the old syntax for now so people can build sourcekit-lsp more easily.

Reply:
I thought we required the latest Swift version to build SourceKit-LSP like we do for SwiftSyntax but I agree that it’s better to be conservative here because it’s reasonable for people to build SourceKit-LSP on its own.